src/ir_Technibel.cpp

```diff
@@ -0,0 +1,416 @@
+  switch (speed) {
+    case stdAc::fanspeed_t::kMin:
+    case stdAc::fanspeed_t::kLow:
+      return kTechnibelAcFanLow;
+    case stdAc::fanspeed_t::kMedium:
+      return kTechnibelAcFanMedium;
+    case stdAc::fanspeed_t::kHigh:
+    case stdAc::fanspeed_t::kMax:
+      return kTechnibelAcFanHigh;
+    default:
+      return kTechnibelAcFanLow;
+  }
+}
+
+// Convert a native fan speed to it's common equivalent.
+stdAc::fanspeed_t IRTechnibelAc::toCommonFanSpeed(const uint8_t speed) {
+  switch (speed) {
+    case kTechnibelAcFanHigh:   return stdAc::fanspeed_t::kHigh;
+    case kTechnibelAcFanMedium: return stdAc::fanspeed_t::kMedium;
+    case kTechnibelAcFanLow:    return stdAc::fanspeed_t::kLow;
+    default:                    return stdAc::fanspeed_t::kLow;
+  }
+}
+
+uint8_t IRTechnibelAc::getMode(void) {
+  uint8_t mode = GETBITS64(remote_state, kTechnibelAcModeOffset,
+                            kTechnibelAcModeSize);
+  return reverseBits(mode, kTechnibelAcModeSize);
+}
+
+void IRTechnibelAc::setMode(const uint8_t mode) {
+  switch (mode) {
+    case kTechnibelAcHeat:
+    case kTechnibelAcFan:
+    case kTechnibelAcDry:
+    case kTechnibelAcCool:
+      break;
+    default:
+      this->setMode(kTechnibelAcCool);
+      return;
+  }
+  uint8_t mode_reversed = reverseBits(mode, kTechnibelAcModeSize);
+  setBits(&remote_state, kTechnibelAcModeOffset, kTechnibelAcModeSize,
+          mode_reversed);
+  setFan(getFan());  // Re-force any fan speed constraints.
+  // Restore previous temp settings for cool mode.
+  setTemp(_saved_temp, _saved_temp_units);
+}
+
+// Convert a standard A/C mode into its native mode.
+uint8_t IRTechnibelAc::convertMode(const stdAc::opmode_t mode) {
+  switch (mode) {
+    case stdAc::opmode_t::kCool:
+      return kTechnibelAcCool;
+    case stdAc::opmode_t::kHeat:
+      return kTechnibelAcHeat;
+    case stdAc::opmode_t::kDry:
+      return kTechnibelAcDry;
+    case stdAc::opmode_t::kFan:
+      return kTechnibelAcFan;
+    default:
+      return kTechnibelAcCool;
+  }
+}
+
+// Convert a native mode to it's common equivalent.
+stdAc::opmode_t IRTechnibelAc::toCommonMode(const uint8_t mode) {
+  switch (mode) {
+    case kTechnibelAcCool:  return stdAc::opmode_t::kCool;
+    case kTechnibelAcHeat:  return stdAc::opmode_t::kHeat;
+    case kTechnibelAcDry:   return stdAc::opmode_t::kDry;
+    case kTechnibelAcFan:   return stdAc::opmode_t::kFan;
+    default:                return stdAc::opmode_t::kAuto;
+  }
+}
+
+void IRTechnibelAc::setSwing(const bool on) {
+  setBit(&remote_state, kTechnibelAcSwingBit, on);
+}
+
+bool IRTechnibelAc::getSwing(void) {
+  return GETBIT64(remote_state, kTechnibelAcSwingBit);
+}
+
+// Convert a standard A/C swing into its native swing.
+bool IRTechnibelAc::convertSwing(const stdAc::swingv_t swing) {
+  switch (swing) {
+    case stdAc::swingv_t::kOff:
+      return false;
+    case stdAc::swingv_t::kAuto:
+      return true;
```

Comment:
This clause is redundant. If you remove it, you'll get the same result due to the `default:` clause.